lib/gpu/lib/src/render_pass.dart

```diff
@@ -7,58 +7,127 @@
+    this.storeAction = StoreAction.store,
+    this.clearValue = const ui.Color(0x00000000),
+    required this.texture,
+    this.resolveTexture = null,
+  });
 
   LoadAction loadAction;
   StoreAction storeAction;
-  ui.Color clearColor;
+  ui.Color clearValue;
+
   Texture texture;
   Texture? resolveTexture;
 }
 
-base class StencilAttachment {
-  StencilAttachment(
-      {this.loadAction = LoadAction.clear,
-      this.storeAction = StoreAction.dontCare,
-      this.clearStencil = 0,
-      required this.texture});
+base class DepthStencilAttachment {
+  DepthStencilAttachment({
+    this.depthLoadAction = LoadAction.clear,
+    this.depthStoreAction = StoreAction.dontCare,
+    this.depthClearValue = 0.0,
+    this.stencilLoadAction = LoadAction.clear,
+    this.stencilStoreAction = StoreAction.dontCare,
+    this.stencilClearValue = 0,
+    required this.texture,
+  });
+
+  LoadAction depthLoadAction;
+  StoreAction depthStoreAction;
+  double depthClearValue;
+
+  LoadAction stencilLoadAction;
+  StoreAction stencilStoreAction;
+  int stencilClearValue;
 
-  LoadAction loadAction;
-  StoreAction storeAction;
-  int clearStencil;
   Texture texture;
 }
 
-/// A descriptor for RenderPass creation. Defines the output targets for raster
-/// pipelines.
-base class RenderTarget {}
+base class ColorBlendEquation {
+  ColorBlendEquation({
+    this.colorBlendOperation = BlendOperation.add,
+    this.sourceColorBlendFactor = BlendFactor.one,
+    this.destinationColorBlendFactor = BlendFactor.oneMinusSourceAlpha,
+    this.alphaBlendOperation = BlendOperation.add,
+    this.sourceAlphaBlendFactor = BlendFactor.one,
+    this.destinationAlphaBlendFactor = BlendFactor.oneMinusSourceAlpha,
+  });
+
+  BlendOperation colorBlendOperation;
+  BlendFactor sourceColorBlendFactor;
+  BlendFactor destinationColorBlendFactor;
+
+  BlendOperation alphaBlendOperation;
+  BlendFactor sourceAlphaBlendFactor;
+  BlendFactor destinationAlphaBlendFactor;
+}
+
+base class SamplerOptions {
+  SamplerOptions({
+    this.minFilter = MinMagFilter.nearest,
+    this.magFilter = MinMagFilter.nearest,
+    this.mipFilter = MipFilter.nearest,
+    this.widthAddressMode = SamplerAddressMode.clampToEdge,
+    this.heightAddressMode = SamplerAddressMode.clampToEdge,
+  });
+
+  MinMagFilter minFilter;
+  MinMagFilter magFilter;
+  MipFilter mipFilter;
+  SamplerAddressMode widthAddressMode;
+  SamplerAddressMode heightAddressMode;
+}
+
+base class RenderTarget {
+  const RenderTarget(
+      {this.colorAttachments = const <ColorAttachment>[],
+      this.depthStencilAttachment});
+
+  factory RenderTarget.singleColor(
+      {ColorAttachment? colorAttachment,
+      DepthStencilAttachment? depthStencilAttachment}) {
+    List<ColorAttachment> colors = [];
+    if (colorAttachment != null) {
+      colors.add(colorAttachment);
+    }
+    return new RenderTarget(
+        colorAttachments: colors,
+        depthStencilAttachment: depthStencilAttachment);
+  }
```

Comment:
With Dart's limited const expression support you won't be able to make this const, but it doesn't need to be a factory either.

It seems odd that the colorAttachment argument to this object is nullable?
```suggestion
    return RenderTarget(
        colorAttachments: [
        
        ],
        depthStencilAttachment: depthStencilAttachment);
  }
```